src/sql/src/plan/query.rs

```diff
@@ -1265,6 +1268,69 @@ fn plan_set_expr(
             let (expr, scope) = plan_nested_query(qcx, query)?;
             Ok((expr, scope))
         }
+        SetExpr::Show(stmt) => {
+            // Some SHOW statements are a SELECT query. Others are various
+            // Plans, of which this supports any that produces Rows directly.
+            // Convert both of these to the needed Hir and Scope.
+
+            fn to_hirscope(
+                plan: Plan,
+                desc: StatementDesc,
+            ) -> Result<(HirRelationExpr, Scope), PlanError> {
+                if let Plan::SendRows(plan) = plan {
+                    let rows = plan
+                        .rows
+                        .iter()
+                        .map(|row| row.iter().collect::<Vec<_>>())
+                        .collect::<Vec<_>>();
+                    let desc = desc.relation_desc.expect("must exist");
+                    let expr = HirRelationExpr::constant(rows, desc.typ().clone());
+                    let scope = Scope::from_source(None, desc.iter_names());
+                    Ok((expr, scope))
+                } else {
+                    sql_bail!("unsupported SHOW statement")
+                }
+            }
+
+            match stmt.as_ref().clone() {
+                Statement::ShowColumns(stmt) => show::show_columns(&qcx.scx, stmt)?.plan_hir(qcx),
+                Statement::ShowCreateConnection(stmt) => to_hirscope(
+                    show::plan_show_create_connection(&qcx.scx, stmt.clone())?,
```

Comment:
ISTM like these `plan_show_create_FOO` functions ought to be changed to return a `SendRowsPlan` directly, so that `to_hirscope` doesn't need to match out the `Plan::SendRows`. Then the compiler can make sure the types line up.

Reply:
Done.